### DISCUSSION TITLE
How to find out if there is any metadata in a table

### DISCUSSION THREAD

**Discussion by hyanwong** [tskit-dev/tskit#2806] [Q&A] [answered]
Is there an efficient way of finding out if a table has any metadata? Or do I have to do `len(ts.tables.nodes.metadata) == 0`? I assume this will be relatively inefficient until https://github.com/tskit-dev/tskit/issues/760 is fixed?

**Comment by petrelharp** [comment]
Not quite the same, but you could check if there's a schema?

**Comment by benjeffery** [comment, marked as answer]
The `repr_html` does `len(table.metadata) > 0` 
Why are you worried about efficiency? I doubt this will be checked in a tight loop?

**Reply by hyanwong** [reply]
Yes, you are right, I'm not checking it a lot, so probably easiest to stick with the obvious.
